### PR TITLE
[MOB-5870] add data center configuration

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 JIRA Ticket(s) if any

* [MOB-5870](https://iterable.atlassian.net/browse/MOB-5870)

## ✏️ Description

- creates the IterableDataRegion enum which will inform the native layer
- adds the dataRegion value to the IterableConfig class
- updates the unit tests for the default config values for the added dataRegion value

[MOB-5870]: https://iterable.atlassian.net/browse/MOB-5870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Ayyanchira — COMMENTED** [review]
A serialization part is missing where this config from Typescript will be converted to respective native configs

File: android/src/main/java/com/iterable/reactnative/Serialization.java [hunk 1]
```diff
@@ -198,6 +199,20 @@ static IterableConfig.Builder getConfigFromReadableMap(ReadableMap iterableConte
                 configBuilder.setLogLevel(logLevel);
             }
 
+            if(iterableContextJSON.has("dataRegion")) {
+                int dataRegion = iterableContextJSON.getInt("dataRegion");
```
Comment (Ayyanchira) [inline android/src/main/java/com/iterable/reactnative/Serialization.java:203]: Although enums are represented as integers internally, in java its treated as class. So a direct mapping like this is not possible. [More reading](https://stackoverflow.com/a/5878986)

File: android/src/main/java/com/iterable/reactnative/Serialization.java [hunk 2]
```diff
@@ -198,6 +199,20 @@ static IterableConfig.Builder getConfigFromReadableMap(ReadableMap iterableConte
...
+                switch (dataRegion) {
+                    case 1:
+                        dataRegion = IterableDataRegion.EU;
```
Comment (Ayyanchira) [inline android/src/main/java/com/iterable/reactnative/Serialization.java:206]: 0 is US and 1 is EU in native. Will be good to have an explicit case check for 0. Default can still stay.

File: android/src/main/java/com/iterable/reactnative/Serialization.java [hunk 3]
```diff
@@ -198,6 +199,20 @@ static IterableConfig.Builder getConfigFromReadableMap(ReadableMap iterableConte
...
+                        break;
+                    default:
+                        dataRegion = IterableDataRegion.US;
+                        break;
+                }
+
+                configBuilder.setDataRegion(dataRegion);
```
Comment (Ayyanchira) [inline android/src/main/java/com/iterable/reactnative/Serialization.java:213]:
```suggestion
 int dataRegionInt = iterableContextJSON.getInt("dataRegionInt");
                IterableDataRegion region;
                switch (dataRegionInt) {
                    case 1:
                        region = IterableDataRegion.EU;
                        break;
                    case 2:
                        region = IterableDataRegion.US;
                    default:
                        region = IterableDataRegion.US;
                        break;
                }

                configBuilder.setDataRegion(region);
```

Comment (Ayyanchira) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:213]: Something like this..

**Review by Ayyanchira — COMMENTED** [review]
Added some comments

File: android/src/main/java/com/iterable/reactnative/Serialization.java (hunk 2 above)
Comment (evantk91) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:206]: Makes sense

File: android/src/main/java/com/iterable/reactnative/Serialization.java (hunk 1 above)
Comment (evantk91) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:203]: Isnt the data region in the RN SDK represented as an integer though?

Comment (evantk91) [reply android/src/main/java/com/iterable/reactnative/Serialization.java:203]: Oh I see... In the Android SDK, we need a getEnum function to get the enum value from the string. Thus, in RN SDK, we should set the value of the data region to the endpoint.

**Review by Ayyanchira — APPROVED** [review]
LGTM

**Pull request merged**